src/libsync/capabilities.cpp

```diff
@@ -63,9 +65,28 @@ bool Capabilities::shareResharing() const
     return _capabilities["files_sharing"].toMap()["resharing"].toBool();
 }
 
-QStringList Capabilities::supportedChecksumTypes() const
+QList<QByteArray> Capabilities::supportedChecksumTypesRaw() const
 {
-    return QStringList();
+    return QList<QByteArray>();
+}
+
+QList<QByteArray> Capabilities::supportedChecksumTypes() const
+{
+    auto list = supportedChecksumTypesRaw();
+    QByteArray cfgType = ConfigFile().transmissionChecksum().toLatin1();
+    if (!cfgType.isEmpty()) {
+        list.prepend(cfgType);
+    }
+    return list;
+}
+
+QByteArray Capabilities::preferredChecksumType() const
+{
+    auto list = supportedChecksumTypes();
+    if (list.isEmpty()) {
+        return QByteArray();
+    }
+    return list.first();
 }
 
 }
```

Comment:
Yes. This is mainly intended for testing and will definitely need to be revised as the server API comes into existence and as we figure out how exactly we want to use the checksums.